pandas/core/groupby/groupby.py

```diff
@@ -768,7 +770,17 @@ def _try_cast(self, result, obj, numeric_only=False):
                 # The function can return something of any type, so check
                 # if the type is compatible with the calling EA.
                 try:
+                    tz_dtype = None
```

Comment:
for now let's not nest this under is_extension_array_dtype as its not pretty as written; make it a separate case.

However, am thinking we we need to have a pair of (optional) routines on EA that handle this type of thing, we are tending to do this in more and more places (IIRC recently in msgpack). IOW I think we want something like

```
serialized, dtype = EA._serialize()
original = EA._from_serialize(serialized, dtype)
```
to work generically. Note when I say serialize this doesn't mean bytes on the wire (necessarily), rather going thru a transform (e.g. here we shift to UTC then transform, then need to transform back).

If you can create an issue for discussion about this.

cc @jbrockmendel @TomAugspurger @jorisvandenbossche